### PR TITLE
Add taints and tolerations

### PR DESCRIPTION
**Pull request by cwize1**

### CODE REVIEW

File: services/cloud/cloud.go [hunk 1]
```diff
@@ -65,6 +66,9 @@ type NodeProperties struct {
 	AuthorizerPort *int32 `json:"authorizerPort,omitempty"`
 
 	Certificate *string `json:"certificate,omitempty"`
+
+	// Taints that restricts which VMs can run on this node.
+	Taints []common.Taint `json:"taints"`
```
Comment (bfjelds) [inline services/cloud/cloud.go:71]: omitempty?

File: go.mod [hunk 2]
```diff
@@ -19,3 +19,5 @@ replace (
 	github.com/Azure/go-autorest v11.1.2+incompatible => github.com/Azure/go-autorest/autorest v0.10.0
 	github.com/miekg/dns => github.com/miekg/dns v1.1.25
 )
+
+replace github.com/microsoft/moc => ../moc
```
[the remark this answers is not in the document]
Comment (cwize1) [reply go.mod:23]: Just local testing. Thanks for the catch.

File: services/common/common.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	// How the taint's value is handled.
+	Operator TolerationOperator `json:"operator"`
+	// The key of the taint to tolerate.
+	Key string `json:"key"`
+	// The value to match against the taint's value.
+	// This field is ignored if 'operator' is set to 'Exists'.
+	Value string `json:"value"`
+	// If true, toleration must match a taint. If false, taint may be present
+	// but is not required.
+	Required bool `json:"required,omitempty"`
```
Comment (vincepnguyen) [inline services/common/common.go:28]: If its not set, empty field, is it required or not?

File: services/compute/compute.go [hunk 4]
```diff
@@ -277,6 +278,8 @@ type VirtualMachineProperties struct {
 	VmType VMType `json:"vmType,omitempty"`
 	// Disable High Availability
 	DisableHighAvailability *bool `json:"disableHighAvailability,omitempty"`
+	// Allows VM to run on nodes that have been tainted for a specific workload.
+	NodeTolerations []common.Toleration `json:"nodetolerations"`
```
Comment (bfjelds) [inline services/compute/compute.go:282]: omitempty?

File: services/common/common.go (hunk 3 above)
Comment (cwize1) [reply services/common/common.go]: If it isn't set, this it will default to `false` which I am happy with.

**Pull request merged**